proposals/0046-astro-env.md

```diff
@@ -0,0 +1,366 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+- Start Date: 2024-04-15
+- Reference Issues: https://github.com/withastro/roadmap/issues/837
+- Implementation PR: <!-- leave empty -->
+
+# Summary
+
+Improve DX and security around environment variables in Astro.
+
+# Example
+
+```js
+// astro.config.mjs
+import { defineConfig, envField } from "astro/config"
+
+export default defineConfig({
+  env: {
+    schema: {
+      API_URL: z.static().private().string(),
```

Comment:
Where does `z` come from? It's not shown in the import. For content collections it is inside of `astro:content`, would it be in `astro/config` here?

Reply:
I've wondered where `z` should come from as we use it from more utilities. I could see `z` being an expected utility for form actions as well. Since it could be used from three places, I wonder if we could:
1. Require Zod be explicitly installed in your project
2. Encourage importing `z` from the `astro/zod` module. This module exists today for our virtual modules to access

Reply:
Update: just noticed `z.static()` is referenced here, which is _not_ a core zod property. I would suggest a custom name like `envVar` or `envField` instead of `z`.